### PR TITLE
Better type handling on BQ reads

### PR DESCRIPTION
**Pull request by jklukas**
Avro may pass back a non-String CharSequence and in practice we see it pass back an org.apache.avro.util.Utf8 instance.

This should fix a failure scenario @whd encountered. I expect it should now be possible to use the sink to read from BQ and write to BQ as long as we don't set `bqSelectedFields`.

### CODE REVIEW

File: ingestion-beam/src/main/java/com/mozilla/telemetry/io/Read.java [hunk 1]
```diff
@@ -105,7 +105,7 @@ public BigQueryInput(ValueProvider<String> tableSpec, BigQueryReadMethod method,
             Map<String, String> attributes = new HashMap<>();
             tableSchema.getFields().stream() //
                 .filter(f -> "STRING".equals(f.getType())) //
-                .forEach(f -> attributes.put(f.getName(), (String) record.get(f.getName())));
+                .forEach(f -> attributes.put(f.getName(), record.get(f.getName()).toString()));
```
Comment (jklukas) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/io/Read.java]: `org.apache.avro.util.Utf8` cannot be cast to String, but `toString` looks to do the right thing.

Comment (whd) [reply ingestion-beam/src/main/java/com/mozilla/telemetry/io/Read.java]: Oddly when applying this patch manually and [running a test job](https://console.cloud.google.com/dataflow/jobsDetail/locations/us-west1/jobs/2019-09-17_16_25_24-3216116149197057161?project=moz-fx-data-beam-nonprod-b5f4&folder) I see a null pointer exception instead: 

     java.io.IOException: Failed to advance reader of source: name: "projects/moz-fx-data-beam-nonprod-b5f4/locations/us/streams/CAsSCEVzazB3Ykh6GgJpcigB"
    
    	at org.apache.beam.runners.dataflow.worker.WorkerCustomSources$BoundedReaderIterator.advance(WorkerCustomSources.java:614)
    	at org.apache.beam.runners.dataflow.worker.util.common.worker.ReadOperation$SynchronizedReaderIterator.advance(ReadOperation.java:399)
    	at org.apache.beam.runners.dataflow.worker.util.common.worker.ReadOperation.runReadLoop(ReadOperation.java:194)
    	at org.apache.beam.runners.dataflow.worker.util.common.worker.ReadOperation.start(ReadOperation.java:159)
    	at org.apache.beam.runners.dataflow.worker.util.common.worker.MapTaskExecutor.execute(MapTaskExecutor.java:77)
    	at org.apache.beam.runners.dataflow.worker.BatchDataflowWorker.executeWork(BatchDataflowWorker.java:411)
    	at org.apache.beam.runners.dataflow.worker.BatchDataflowWorker.doWork(BatchDataflowWorker.java:380)
    	at org.apache.beam.runners.dataflow.worker.BatchDataflowWorker.getAndPerformWork(BatchDataflowWorker.java:305)
    	at org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.doWork(DataflowBatchWorkerHarness.java:140)
    	at org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.call(DataflowBatchWorkerHarness.java:120)
    	at org.apache.beam.runners.dataflow.worker.DataflowBatchWorkerHarness$WorkerThread.call(DataflowBatchWorkerHarness.java:107)
    	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    	at java.lang.Thread.run(Thread.java:745)
    Caused by: java.lang.NullPointerException
    	at com.mozilla.telemetry.io.Read$BigQueryInput.lambda$null$1(Read.java:108)
    	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
    	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
    	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1376)
    	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
    	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
    	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
    	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
    	at com.mozilla.telemetry.io.Read$BigQueryInput.lambda$expand$d7d74fec$1(Read.java:108)
    	at org.apache.beam.sdk.io.gcp.bigquery.BigQueryStorageStreamSource$BigQueryStorageStreamReader.readNextRecord(BigQueryStorageStreamSource.java:226)
    	at org.apache.beam.sdk.io.gcp.bigquery.BigQueryStorageStreamSource$BigQueryStorageStreamReader.advance(BigQueryStorageStreamSource.java:203)
    	at org.apache.beam.runners.dataflow.worker.WorkerCustomSources$BoundedReaderIterator.advance(WorkerCustomSources.java:611)
    	... 14 more

Comment (jklukas) [reply ingestion-beam/src/main/java/com/mozilla/telemetry/io/Read.java]: The NPE makes sense. Now that we're calling `.toString()`, we'll throw an NPE if we try to process a field that has a null. Will fix.

**Comment by jklukas** [comment]
I also realized that the `submission_timestamp` field is a TIMESTAMP that's passed back as a Long in the avro record, so we weren't being comprehensive.

I've updated to handle (and test handling of) strings and timestamp fields.

**Pull request merged**